packages/core-js/internals/structured-clone.js

```diff
@@ -0,0 +1,113 @@
+var getOwnPropertyNames = require('../internals/object-get-own-property-names');
+var classof = require('../internals/classof');
+var getBuiltin = require('../internals/get-built-in');
+var isObject = require('../internals/is-object');
+var has = require('../internals/has');
+
+var Set = getBuiltin('Set');
+var Map = getBuiltin('Map');
+
+function createDataCloneError(message) {
+  if (typeof DOMException === 'function') {
+    return new DOMException(message, 'DataCloneError');
+  }
+  return new Error(message);
+}
+
+/**
+ * Tries best to replicate structuredClone behaviour.
+ *
+ * @param {WeakMap} weakmap cache map
+ * @param {any} value object to clone
+ */
+module.exports = function structuredCloneInternal(weakmap, value) {
+  if (isSymbol(value)) throw createDataCloneError('Symbols are not cloneable');
+  if (!isObject(value)) return value;
+  if (value === null) return null;
+  if (weakmap.has(value)) return weakmap.get(value); // effectively preserves circular references
+
+  var cloned, i, deep;
+  var type = classof(value);
+
+  switch (type) {
+    case 'Boolean':
+    case 'BigInt':
+    case 'Number':
+    case 'String':
+      cloned = Object(value.valueOf());
+      break;
+    case 'Date':
+      cloned = new Date(value.valueOf());
+      break;
+    case 'RegExp':
+      cloned = new RegExp(value);
+      break;
+    case 'Map':
+      cloned = new Map();
+      deep = true;
+      break;
+    case 'Set':
+      cloned = new Set();
+      deep = true;
+      break;
+    case 'Error':
+    case 'EvalError':
+    case 'RangeError':
+    case 'ReferenceError':
+    case 'SyntaxError':
+    case 'TypeError':
+    case 'URIError':
+      cloned = value.constructor(value.message.toString());
+      deep = true; // clone stack after storing in the weakmap
+      break;
+    case 'Array':
+      cloned = [];
+      deep = true;
+      break;
+    case 'Object':
+      cloned = {};
+      deep = true;
+      break;
+    default:
+      throw createDataCloneError('Uncloneable type: ' + type);
+  }
+
+  weakmap.set(value, cloned);
+
+  if (deep) switch (type) {
+    case 'Map':
+      value.forEach(function (v, k) {
+        cloned.set(structuredCloneInternal(weakmap, k), structuredCloneInternal(weakmap, v));
+      });
+      break;
+    case 'Set':
+      value.forEach(function (v) {
+        cloned.add(structuredCloneInternal(weakmap, v));
+      });
+      break;
+    case 'Error':
+      // Attempt to clone the stack.
+      if (
+        !has(value, 'stack') && // Chrome, Safari
+        !has(Error.prototype, 'stack') // Firefox
+      ) break;
+      try {
+        cloned.stack = structuredCloneInternal(weakmap, value.stack);
+      } catch (_) {
+        return cloned; // Stack cloning not avaliable.
+      }
+      break;
+    case 'Array':
+    case 'Object':
+      var properties = getOwnPropertyNames.f(value);
+      for (i = 0; i < properties.length; i++) {
+        cloned[structuredCloneInternal(weakmap, properties[i])] =
```

Comment:
`getOwnPropertyNames` returns only primitive strings, so makes no sense to use `structuredCloneInternal` here.